src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/LabelSelectorExpressions.java

```diff
@@ -0,0 +1,83 @@
+package com.cloudbees.jenkins.plugins.kubernetes_credentials_provider;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.fabric8.kubernetes.api.model.LabelSelector;
+import io.fabric8.kubernetes.api.model.LabelSelectorBuilder;
+
+/**
+ * Parser for <a href="https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors">Kubernetes Label Selectors</a>.
+ */
+class LabelSelectorExpressions {
+
+    /**
+     * Parse Kubernetes label selector expression.
+     * Example:
+     * <pre>
+     * app in (jenkins-dev, jenkins-all)
+     * </pre>
+     * @param selector label selector expression or null
+     * @return parsed label selector
+     */
+    @NonNull
+    static LabelSelector parse(String selector) {
+        LabelSelectorBuilder lsb = new LabelSelectorBuilder();
+        if (selector != null && !selector.trim().isEmpty()) {
+            Matcher matcher = Pattern.compile("[^,]+?\\([^()]+\\)|[^,]+").matcher(selector);
+            while (matcher.find()) {
+                String expression = matcher.group();
+                expression = expression.replaceFirst("!=|=", " $0 ");
+                String[] tokens = expression.trim().split("\\s+", 3);
+                switch (tokens.length) {
+                    case 1:
+                        if (tokens[0].startsWith("!")) {
+                            lsb.addNewMatchExpression()
+                                    .withKey(tokens[0].substring(1))
+                                    .withOperator("DoesNotExist")
+                                    .endMatchExpression();
+                        } else {
+                            lsb.addNewMatchExpression()
+                                    .withKey(tokens[0])
+                                    .withOperator("Exists")
+                                    .endMatchExpression();
+                        }
+                        break;
+                    case 2:
```

Comment:
someone has provided a selector statement that is invalid and this is just being swallowed.
At this point I would expect at minimum an ERROR in the logs, but I would suggest that this makes all secrets unavailable from this plugin as it is a missconfiguration that will be harder to spot otherwise (the fact that things are not missing may not be realized until some time after the application of the property)

Reply:
Took your suggestion and throw a parser exception which is caught by the provider which then logs the error and creates an administrative error message.